src/bin/pg_autoctl/Makefile

```diff
@@ -47,23 +49,35 @@ LIBS += -lpq
 
 all: $(PG_AUTOCTL) ;
 
+# Based on Postgres Makefile for automatic dependency generation
+# https://github.com/postgres/postgres/blob/1933ae629e7b706c6c23673a381e778819db307d/src/Makefile.global.in#L890-L924
+%.o : %.c
+	@if test ! -d $(DEPDIR); then mkdir -p $(DEPDIR); fi
+	$(CC) $(CFLAGS) -c -MMD -MP -MF$(DEPDIR)/$(*F).Po -o $@ $<
```

Comment:
Can we add the new flags to a variable and then to CFLAGS, as we use it everywhere we use CFLAGS I believe?

Reply:
I tried to do that, but since these flags contain variables which depend on being in the target it doesn't work, i.e  `$(*F)`. I'm fine with adding the -MMD and -MP flags to CFLAGS though, but maybe it's nicer to keep these flags together.